src/ensembl/src/content/app/species/state/speciesSidebarSlice.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+import { RootState } from 'src/store';
+
+type SpeciesPageSidebarState = {
+  isOpen: boolean;
+};
+
+const initialState: SpeciesPageSidebarState = {
+  isOpen: true
+};
+
+const speciesPageSidebarSlice = createSlice({
+  name: 'species-page-sidebar',
+  initialState,
+  reducers: {
+    toggleSidebar(state) {
+      state.isOpen = !state.isOpen;
+    }
+  }
+});
+
+export const { toggleSidebar } = speciesPageSidebarSlice.actions;
+
+export const isSidebarOpen = (state: RootState) =>
```

Comment:
Do we not put these in a separate selector file?

Reply:
I don't have a particularly strong opinion on this. Redux-toolkit [says](https://redux-toolkit.js.org/usage/usage-guide#creating-slices-of-state) that it favours the "duck" pattern when all redux logic is co-located in a single file:

![image](https://user-images.githubusercontent.com/6834224/91420239-4f129d80-e84c-11ea-96c4-656d0db11319.png)

but at the same time, they don't include selectors in their example. I thought, if the slice is simple, selectors could be put in the same file; but I can move them in a separate file if it helps.

The sidebar slice is very simple at the moment. In EntityViewer, sidebar reducer is keyed by species id. I didn't do it here, hoping that we won't need it, but I suspect that we will...

Reply:
I agree that it is simple enough to leave it here but surely we will be moving it later on when there are more selectors.
I am just thinking that it will be simpler at that point rather than having to deal the move and updating the paths later on.

Reply:
Moved the selector to a separate file.